codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/test/protocol/fakeprotocol/generators/BuilderGenerator.java

```diff
@@ -39,12 +42,24 @@ public BuilderGenerator(GenerationContext context) {
     @Override
     protected void renderOperationBuildMethod(OperationShape operation, Shape inputShape) {
         writer.openBlock("def self.build(http_req, input:)");
-        if (Streaming.isStreaming(model, inputShape)) {
-            renderStreamingBodyBuilder(inputShape);
+
+        // checks for Payload member
+        Optional<MemberShape> httpPayloadMember = inputShape.members()
+                .stream()
+                .filter((m) -> m.hasTrait(HttpPayloadTrait.class))
+                .findFirst();
+        if (httpPayloadMember.isPresent()) {
+            if (Streaming.isStreaming(model, inputShape)) {
+                renderStreamingBodyBuilder(inputShape);
+            } else {
+                writer.write("http_req.body = StringIO.new(input[:$L] || '')",
```

Comment:
We should add a comment that this will only work for String/Blob/Number data types which is sufficent for testing but not complete.  (the full solution woudl be to create a body serializer visitor here and provide implementations for all of the smithy types as we do in the RestBuilderGeneratorBase).

Reply:
Are we coming back to do this later?

Reply:
I dont think theres any reason for now to do so since this is just used as a minimal test protocol (ie, we don't actually serialize anything normally.... but its useful for testing some features.)